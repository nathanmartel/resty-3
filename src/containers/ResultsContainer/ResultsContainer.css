.resultsContainer {
  background-color: #eee;
  padding: 1.5rem;
  width: 100%;
}

.error {
  color: red;
}