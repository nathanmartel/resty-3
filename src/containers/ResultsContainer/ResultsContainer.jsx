import React from 'react';
import Results from '../../components/Results/Results';
import { useRes, useLoading, useError } from '../../hooks/AppProvider/AppProvider';
import styles from './ResultsContainer.css';


const ResultsContainer = () => {

  const res = useRes();
  const loading = useLoading();
  const error = useError();

  const resultsObj = res.map((item, index) => <Results key={index} item={item}/>);

  return (
    <>
      <div className={styles.resultsContainer}>
        <h3>Results</h3>
        <hr />
        { loading ? (
          <p>Loading...</p> 
        ) : (
          <>{resultsObj}</>
        )}
        { error && <p className={styles.error}>{error}</p> }
      </div>
    </>
  );
};

export default ResultsContainer;
