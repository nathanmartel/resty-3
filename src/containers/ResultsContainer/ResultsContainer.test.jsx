import React from 'react';
import { shallow } from 'enzyme';
import ResultsContainer from './ResultsContainer';

describe('ResultsContainer component', () => {
  it('renders ResultsContainer', () => {
    const res = [{
      test: 'test'
    }];
    const wrapper = shallow(<ResultsContainer res={res} loading={false} error={''} />);
    expect(wrapper).toMatchSnapshot();
  });
});
