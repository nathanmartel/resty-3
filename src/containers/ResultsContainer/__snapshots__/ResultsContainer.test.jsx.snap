// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsContainer component renders ResultsContainer 1`] = `
<Fragment>
  <div
    className="resultsContainer"
  >
    <h3>
      Results
    </h3>
    <hr />
    <Results
      item={
        Object {
          "test": "test",
        }
      }
      key="0"
    />
  </div>
</Fragment>
`;
